From 7e51aa7b095bc7cd41970d0be258de688531e2e6 Mon Sep 17 00:00:00 2001
From: Jiyong Park <jiyong@google.com>
Date: Fri, 24 Feb 2017 17:49:42 +0900
Subject: [PATCH 4/8] msm8974: Move QCOM HALs to vendor partition
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In order to build generic system image that is agnostic to SoC or
device, such HALs should be moved from system partition to vendor
partition.

HALs moved: copybit, gralloc, hwcomposer, light, memtrack, hdmi_cec

Bug: 34826389
Bug: 35232083
Test: build & run and check whether the HAL is loaded from
/vendor/lib[64]/hw.
$ su; lsof | grep hwcomposer

Change-Id: I2bb617ac71ab040a5b1ba306c194e8672ac84561
Keep msm8974 in sync with the rest, post removal
Signed-off-by: D. Andrei Măceș <Andrei.Maces@alumni.nd.edu>
---
 msm8974/libcopybit/Android.mk    | 3 ++-
 msm8974/libgralloc/Android.mk    | 3 ++-
 msm8974/libhwcomposer/Android.mk | 3 ++-
 msm8974/liblight/Android.mk      | 3 ++-
 msm8974/libmemtrack/Android.mk   | 3 ++-
 5 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/msm8974/libcopybit/Android.mk b/msm8974/libcopybit/Android.mk
index 7ec99daf3..0e494af4a 100644
--- a/msm8974/libcopybit/Android.mk
+++ b/msm8974/libcopybit/Android.mk
@@ -22,7 +22,8 @@ include $(BUILD_COPY_HEADERS)
 
 include $(CLEAR_VARS)
 LOCAL_MODULE                  := copybit.$(TARGET_BOARD_PLATFORM)
-LOCAL_MODULE_PATH             := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH    := hw
+LOCAL_PROPRIETARY_MODULE      := true
 LOCAL_MODULE_TAGS             := optional
 LOCAL_C_INCLUDES              := $(common_includes) $(kernel_includes)
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libdl libmemalloc
diff --git a/msm8974/libgralloc/Android.mk b/msm8974/libgralloc/Android.mk
index 2f18ee894..956302c04 100644
--- a/msm8974/libgralloc/Android.mk
+++ b/msm8974/libgralloc/Android.mk
@@ -21,7 +21,8 @@ include $(CLEAR_VARS)
 LOCAL_CLANG_CFLAGS            += -Wno-gnu-designator
 
 LOCAL_MODULE                  := gralloc.$(TARGET_BOARD_PLATFORM)
-LOCAL_MODULE_PATH             := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH    := hw
+LOCAL_PROPRIETARY_MODULE      := true
 LOCAL_MODULE_TAGS             := optional
 LOCAL_C_INCLUDES              := $(common_includes) $(kernel_includes)
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libmemalloc
diff --git a/msm8974/libhwcomposer/Android.mk b/msm8974/libhwcomposer/Android.mk
index 951e1700c..d0038039b 100644
--- a/msm8974/libhwcomposer/Android.mk
+++ b/msm8974/libhwcomposer/Android.mk
@@ -5,7 +5,8 @@ include $(CLEAR_VARS)
 # hwc.cpp uses GNU old-style field designator extension.
 LOCAL_CLANG_CFLAGS            += -Wno-gnu-designator
 LOCAL_MODULE                  := hwcomposer.$(TARGET_BOARD_PLATFORM)
-LOCAL_MODULE_PATH             := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH    := hw
+LOCAL_PROPRIETARY_MODULE      := true
 LOCAL_MODULE_TAGS             := optional
 LOCAL_C_INCLUDES              := $(common_includes) $(kernel_includes)
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libEGL liboverlay \
diff --git a/msm8974/liblight/Android.mk b/msm8974/liblight/Android.mk
index 40115d5b7..511157800 100644
--- a/msm8974/liblight/Android.mk
+++ b/msm8974/liblight/Android.mk
@@ -18,7 +18,8 @@ LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := lights.c
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH    := hw
+LOCAL_PROPRIETARY_MODULE      := true
 LOCAL_SHARED_LIBRARIES := liblog
 LOCAL_CFLAGS := $(common_flags) -DLOG_TAG=\"qdlights\"
 LOCAL_MODULE := lights.$(TARGET_BOARD_PLATFORM)
diff --git a/msm8974/libmemtrack/Android.mk b/msm8974/libmemtrack/Android.mk
index 306eb07a9..24f8abd53 100644
--- a/msm8974/libmemtrack/Android.mk
+++ b/msm8974/libmemtrack/Android.mk
@@ -18,7 +18,8 @@ LOCAL_PATH := $(call my-dir)
 # hw/<POWERS_HARDWARE_MODULE_ID>.<ro.hardware>.so
 include $(CLEAR_VARS)
 
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH    := hw
+LOCAL_PROPRIETARY_MODULE      := true
 LOCAL_C_INCLUDES += hardware/libhardware/include
 LOCAL_SHARED_LIBRARIES := liblog
 LOCAL_SRC_FILES := memtrack_msm.c kgsl.c
-- 
2.17.1

